influxql/select.go

```diff
@@ -132,12 +132,15 @@ func buildFieldIterators(fields Fields, ic IteratorCreator, opt IteratorOptions)
 			// Build iterators for calls first and save the iterator.
 			// We do this so we can keep the ordering provided by the user, but
 			// still build the Call's iterator first.
-			if ContainsVarRef(f.Expr) {
+
+			expr := Reduce(f.Expr, nil)
```

Comment:
This should be after the `ContainsVarRef` call to avoid any unnecessary reductions in the case of an aux field. You also need to add this reduction when building the aux iterators.